build/deploy/core-service.libsonnet

```diff
@@ -1,11 +1,56 @@
 local base = import 'base.libsonnet';
 local volumes = import 'volumes.libsonnet';
 
+local ingress(metadata) = base.Ingress(metadata, 'https-ingress') {
+  metadata+: {
+    annotations: {
+      'kubernetes.io/ingress.global-static-ip-name': metadata.backend.ipName,
+      'kubernetes.io/ingress.allow-http': 'false',
+    },
+  },
+  spec: {
+    defaultBackend: {
+      service: {
+        name: 'core-service',
+        port: {
+          number: metadata.backend.port,
+        }
+      }
+    },
+  },
+};
+
 {
+  ManagedCertIngress(metadata): {
+    ingress: ingress(metadata) {
+      metadata+: {
+        annotations+: {
+          'networking.gke.io/managed-certificates': 'https-certificate',
+        },
+      },
+    },
+    managedCert: base.ManagedCert(metadata, 'https-certificate') {
+      spec: {
+        domains: [
+          metadata.backend.hostname,
+        ],
+      },
+    },
+  },
+
+  PresharedCertIngress(metadata, certName): ingress(metadata) {
+    metadata+: {
+      annotations+: {
+        'ingress.gcp.kubernetes.io/pre-shared-cert': certName,
+      },
+    },
+  },
+
   all(metadata): {
     service: base.Service(metadata, 'core-service') {
```

Comment:
I think you're missing `ingress: $.ManagedCertIngress(metadata)` here, there's not Ingress in the rendered templates

Reply:
Very good catch, thanks